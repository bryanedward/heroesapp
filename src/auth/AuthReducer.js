import { types } from "../types/types";

const loginAuth = {
    types: types.login,
    payload: {
        name:'edward',
        password: '123'
    }
}

export const AuthReducer = ( state ={}, action) => {
    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }            
            break;
        case types.logout:
            return {
                logged: false
            }
            break
    
        default:
            return state;
    }
}