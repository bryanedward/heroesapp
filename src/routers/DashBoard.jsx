import React from 'react'
import { Routes, Route, Link } from "react-router-dom";
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { Navbar } from '../components/ui/Navbar';
import { DcScreen} from '../components/dc/DcScreen'
import { HeroScreen } from '../components/hero/HeroScreen';


export const DashBoard = () => {
  return (
    <>
        <Navbar/>
            <h1>marvel app!</h1>
        <Routes>
                <Route path="/" element={<MarvelScreen />} />
                <Route path="/marvel" element={<MarvelScreen />} />
                <Route path="/dc" element={<DcScreen />} />
                <Route path='/heroe' element={<HeroScreen/>}/>
            </Routes>
    </>
  )
}
