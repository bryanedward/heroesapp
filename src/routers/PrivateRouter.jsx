import React, { useContext } from 'react'
import { Navigate } from 'react-router-dom'
import { AuthContext } from '../auth/AuthContext'
import { MarvelScreen } from '../components/marvel/MarvelScreen'

export const PrivateRouter = ({children}) => {
const {state} = useContext(AuthContext)
return (
    state.logged ? children : <Navigate to='/login'/>
  )
}
