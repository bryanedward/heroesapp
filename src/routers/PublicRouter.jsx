import React, { useContext } from 'react'
import { Navigate } from 'react-router-dom'
import { AuthContext } from '../auth/AuthContext'
import { LoginScreen } from '../components/login/LoginScreen'

export const PublicRouter = () => {
    const {state} = useContext(AuthContext)
  return ( state.logged ? <Navigate to='/marvel'/> : <LoginScreen/>
  )
}
