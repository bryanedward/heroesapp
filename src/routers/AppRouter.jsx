import React from 'react'
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { DashBoard } from './DashBoard';
import { PrivateRouter } from './PrivateRouter';
import { PublicRouter } from './PublicRouter';


export const AppRouter = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/login' element={<PublicRouter/>}/>
                <Route path='/*' element={<PrivateRouter>
                    <DashBoard/>
                </PrivateRouter>}/>
            </Routes>
            
        </BrowserRouter>
    )
}
