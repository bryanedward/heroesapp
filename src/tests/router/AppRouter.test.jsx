import { mount } from 'enzyme'
import { AuthContext } from '../../auth/AuthContext'
import { AppRouter } from '../../routers/AppRouter'



describe('app router', () => {
    test('validar contenido del componente', () => {
        const contextValue = {
            state: {
                logged: false
            }
        }
        const wrapper = mount(<AuthContext.Provider value={contextValue}>
            <AppRouter />
        </AuthContext.Provider>)
        expect(wrapper.find('h4').text().trim()).toBe('login')
    })
    test('validar el componentes de marvelScreeen', () => {
        const contextValue = {
            state: {
                name: 'dev',
                logged: true
            }
        }
        const wrapper = mount(<AuthContext.Provider value={contextValue}>
            <AppRouter />
        </AuthContext.Provider>)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.navbar').exists()).toBe(true)
    })
})