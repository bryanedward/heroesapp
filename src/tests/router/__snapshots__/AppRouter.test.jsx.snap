// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app router validar el componentes de marvelScreeen 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      HeroApps
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-linkactive"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/heroe"
          onClick={[Function]}
        >
          Heroe
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          dev
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div>
    <div
      className="d-flex flex-wrap gap-5"
    >
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-batman.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Batman
        </p>
        <a
          href="/heroe/dc-batman"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-superman.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Superman
        </p>
        <a
          href="/heroe/dc-superman"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-flash.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Flash
        </p>
        <p
          className="text-muted"
        >
          Jay Garrick, Barry Allen, Wally West, Bart Allen
        </p>
        <a
          href="/heroe/dc-flash"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-green.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Green Lantern
        </p>
        <p
          className="text-muted"
        >
          Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
        </p>
        <a
          href="/heroe/dc-green"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-arrow.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Green Arrow
        </p>
        <a
          href="/heroe/dc-arrow"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-wonder.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Wonder Woman
        </p>
        <a
          href="/heroe/dc-wonder"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-martian.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Martian Manhunter
        </p>
        <p
          className="text-muted"
        >
          Martian Manhunter
        </p>
        <a
          href="/heroe/dc-martian"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-robin.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Robin/Nightwing
        </p>
        <a
          href="/heroe/dc-robin"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-blue.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Blue Beetle
        </p>
        <p
          className="text-muted"
        >
          Dan Garret, Ted Kord, Jaime Reyes
        </p>
        <a
          href="/heroe/dc-blue"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/dc-black.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Black Canary
        </p>
        <p
          className="text-muted"
        >
          Dinah Drake, Dinah Lance
        </p>
        <a
          href="/heroe/dc-black"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-spider.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Spider Man
        </p>
        <a
          href="/heroe/marvel-spider"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-captain.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Captain America
        </p>
        <a
          href="/heroe/marvel-captain"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-iron.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Iron Man
        </p>
        <a
          href="/heroe/marvel-iron"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-thor.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Thor
        </p>
        <a
          href="/heroe/marvel-thor"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-hulk.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Hulk
        </p>
        <a
          href="/heroe/marvel-hulk"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-wolverine.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Wolverine
        </p>
        <a
          href="/heroe/marvel-wolverine"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-daredevil.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Daredevil
        </p>
        <a
          href="/heroe/marvel-daredevil"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-hawkeye.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Hawkeye
        </p>
        <a
          href="/heroe/marvel-hawkeye"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-cyclops.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Cyclops
        </p>
        <a
          href="/heroe/marvel-cyclops"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
      <div
        className="card"
        style={
          Object {
            "width": "13pc",
          }
        }
      >
        <img
          alt=""
          src="assets/marvel-silver.jpg"
          style={
            Object {
              "objectFit": "cover",
              "width": "100%",
            }
          }
        />
        <p>
          Silver Surfer
        </p>
        <a
          href="/heroe/marvel-silver"
          onClick={[Function]}
        >
          ver mas...
        </a>
      </div>
    </div>
  </div>,
]
`;
